zerver/lib/events.py

```diff
@@ -322,6 +323,18 @@ def fetch_initial_state_data(
         state["max_topic_length"] = MAX_TOPIC_NAME_LENGTH
         state["max_message_length"] = settings.MAX_MESSAGE_LENGTH
 
+    if want("realm_user_settings_defaults"):
+        realm_user_default = RealmUserDefault.objects.get(realm=realm)
+        state["realm_user_settings_defaults"] = {}
+        for property_name in RealmUserDefault.property_types:
+            state["realm_user_settings_defaults"][property_name] = getattr(
+                realm_user_default, property_name
+            )
+        for property_name in RealmUserDefault.notification_setting_types:
+            state["realm_user_settings_defaults"][property_name] = getattr(
+                realm_user_default, property_name
+            )
+
```

Comment:
Here also we would require `available_notification_sounds` and `emojiset_choices` that we sent currently in `user_settings` object. But that should be same in both cases - for both user and realm settings. Do we want to duplicate them here or just remove it from `user_settings` object and include in the top-level object.

Reply:
I'm not sure what the best approach is here; maybe start an thread in `#api design`?

Reply:
On thinking again, I think we would want to have two copies, as there can be case when `event_types` only includes `realm_user_settings_defaults` and also it would not be of use if none of `realm_user_settings_defaults` and `user_settings` is included in event types.

Reply:
Sounds reasonable.